### DISCUSSION TITLE
Activity timeout reset of AmmoWorld?

### DISCUSSION THREAD

**Discussion by entangledothers** [cables-gl/cables_docs#342] [Answers & Questions]
Is there a way of resetting `AmmoWorld` when there has been X number of seconds without input? I have been trying with `InteractionListener` connected to the `AmmoWorld` reset via a `TriggerOnChangeBoolean`, which works, but **only** once the mouse moves again after inactivity (so the reset only updates the `AmmoWorld` once the mouse moves again).

Is there a better/proper way of having an `AmmoWorld` reset after an inactivity timeout? _(not sure if the reset only upon new activity is a bug, happy to change category of this issue if it is)_

**Comment by pandrr** [comment]
maybe you are looking for https://cables.gl/op/Ops.Time.TimeSinceTrigger ?

**Reply by entangledothers** [reply]
That appears to be what I was looking for, but wiring it up as the example patch seems to indicate causes the character to freeze after the specified time until the canvas has been out and in of focus upon which it crashes? [See this patch for an example ](https://dev.cables.gl/p/631ef8154239227b62351041?s=n9clCb3zJTSZu27s). Am I wiring it up in the wrong way?

**Comment by pandrr** [comment]
will look into crash, but why do you want to reset it ?

**Reply by entangledothers** [reply]
As the patch is part of an exhibition, it's ideal if the experience "resets" after a certain amount of time so that new viewers start consistently from the "starting point" of the work, if that makes sense?

**Reply by steam0r** [reply]
that sounds more like a job for https://dev.cables.gl/op/Ops.Trigger.Interval

**Reply by entangledothers** [reply]
Ah, I meant that the reset should occur X seconds after input is "idle", this doesn't seem to be the case with Ops.Trigger.Interval?
